osquery/events/linux/bpf/systemstatetracker.cpp

```diff
@@ -0,0 +1,1017 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <iomanip>
+#include <sstream>
+
+#include <osquery/events/linux/bpf/systemstatetracker.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/status/status.h>
+
+#include <linux/fcntl.h>
+#include <linux/netlink.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+namespace osquery {
+struct SystemStateTracker::PrivateData final {
+  Context context;
+  IProcessContextFactory::Ref process_context_factory;
+};
+
+SystemStateTracker::Ref SystemStateTracker::create() {
+  IProcessContextFactory::Ref process_context_factory;
+  auto status = IProcessContextFactory::create(process_context_factory);
+  if (!status) {
+    throw status;
+  }
+
+  return create(std::move(process_context_factory));
+}
+
+SystemStateTracker::Ref SystemStateTracker::create(
+    IProcessContextFactory::Ref process_context_factory) {
+  try {
+    return SystemStateTracker::Ref(
+        new SystemStateTracker(std::move(process_context_factory)));
+
+  } catch (const Status& status) {
+    LOG(ERROR) << status.getMessage();
```

Comment:
I generally do not like this pattern, I think we this should add a tiny bit more context:

```
LOG(ERROR) << "Could not create state tracker: " << status.getMessage();
```

Reply:
> Also it looks like the exit code is a signed value and the table is reporting as unsigned (just a guess from observation).

Yes, the library behind always work with std::uint64_t values as it's the most generic type (which also works well when pointers are returned by the traced functions).